### DISCUSSION TITLE
`@QuarkusIntegrationTest` environment variables and container name

### DISCUSSION THREAD

**Discussion by andrewazores** [quarkusio/quarkus#33806] [Q&A] [answered]
Hi,

I have some existing integration tests in my project that I am trying to carry forward as I port it to Quarkus. The general test setup is similar enough that the application starts and runs fine, but:

1. I would like to be able to set some environment variables for the test scenario, for example `JAVA_OPTS_APPEND`. Is there a way to specify environment variables that should be set when the Quarkus integration test runner does the `podman run` command to start the application container?
2. Is there a way to retrieve the container's name or hostname from the test code? It's a bit involved to explain exactly why I need this, but it would be convenient if I could get the container network internal hostname (ex. `f3274944e014` from my last attempt) somehow from within the test. Basically, I need this so I can use it in a request to the application and tell it to make a request back to itself. I know I can query the Podman API to get the container hostname, but then from the test code I would also need to know the container's name or ID so I can query Podman for further information about that particular container. Or is the best way to simply look for containers with names matching the pattern `quarkus-integration-test-*`?

Thanks in advance!

[a comment by an automated account is not shown]

**Reply by andrewazores** [reply]
Alternatively, if there is a way to customize the `@QuarkusIntegrationTest` `podman run`, then I think I can also solve my problem if I am able to set `--hostname` and `--labels` flags.

**Comment by geoand** [comment]
Hi,

For `1`: Do you really need to use environment variables for the container being launched? What exactly is it you are trying to configure?

For `2`: You can control which network the container will be launch on with `quarkus.test.container.network`

**Reply by andrewazores** [reply]
For `1`: this is for Cryostat, so I am trying to use `JAVA_OPTS_APPEND` to set JVM flags so that I can turn on JMX. I can get around it by just turning those flags on in the `Dockerfile.jvm` for a test-specific image, but I'd prefer to have the same image built with the flags off by default and turn them on on-demand.
For `2`: I will try to experiment with this, thanks for the tip. I'm not sure if it will work in my case reviewing [the doc](https://quarkus.io/guides/all-config#quarkus-core_quarkus.test.container.network), since I do also have DevServices running in the test.

**Reply by geoand** [reply]
We can certainly support passing environment variables if needed. Just let me know 😉

**Reply by andrewazores** [reply]
That would be quite helpful. Would it also be possible to pass `podman run --label foo=bar`? We have a `podman-in-podman` kind of functionality in Cryostat where it queries `podman.sock` for `libpod` data about running containers, and uses labels to filter the results. This would mean that the tests could exercise this functionality, and the tests could also use the labels to identify containers "from the outside" (tying into my original question 2).

**Reply by andrewazores** [reply]
I worked around the `--label` for now by just adding a `LABEL` directive in my Dockerfile. This isn't so bad, so if `--label` isn't an acceptable feature for `@QuarkusIntegrationTest`, I think I can get by. Just being able to set env vars for `JAVA_OPTS_APPEND` is the last significant hurdle for my case.

**Reply by geoand** [reply]
I think we can do both. I'll have a look this week

**Reply by geoand** [reply, marked as answer]
[Here](https://github.com/quarkusio/quarkus/pull/33840) it is

**Comment by andrewazores** [comment]
Thank you @geoand !

**Reply by geoand** [reply]
YW!
